app/controllers/all_galleries.py

```diff
@@ -3,13 +3,78 @@
 from werkzeug.security import check_password_hash
 from flask import session
 from app.models.Forms import *
+from app.models.Galleries import *
+from datetime import datetime
+
+
 
 @app.route('/', methods=["GET","POST"])
 def all_galleries():
-    user = Forms.get(fid=1)
-    name = user.first_name
-    print(name)
-    print(doesUserHaveRole("admin"))
-    return render_template('views/all_galleries.html')
+  
+    formGalleryIds=[] 
+    gallaryIds = []
+    
+    
+    map1={} #maps gallery ids and titles
+    map2={} #maps gallary to the total numbe of submissions it has
+    map3 = {} #maps gallery to its status e.g closed,open,active
+    map4 = {} #Stores final dates for each gallery depending on its status.
+    
+    
+    
+    
+    
+    for gallery in Galleries.select():
+        map1[gallery.gid]=gallery.title
+        gallaryIds.append(gallery.gid)
+        
+    for form in Forms.select():
+        formGalleryIds.append(form.gallery_id)
+        
+    
+    #initializing gallery submission count
+    for gallerycount in gallaryIds:
+        map2[gallerycount] = 0
+        
+    
+    #Counting number of submissions in each gallery
+    for galleryid in formGalleryIds:
+        if galleryid in map2:
+            map2[galleryid] +=1
+            
+    
+    today = datetime.now() #For purposes of determining the status of a galler
+    for gallery_name in map2:
+        #getting dates for each gallery
+        obj=Galleries.get(gid=gallery_name)
+        opened= obj.open_date
+        closed=obj.close_date
+        
+        
+        if today >= opened and today >= closed:
```

Comment:
We should add this function to the model for each instance.

- [ ] Add method to instance to get status